bundler/lib/bundler/lockfile_generator.rb

```diff
@@ -69,10 +69,11 @@ def add_dependencies
     def add_checksums
       out << "\nCHECKSUMS\n"
 
-      empty_store = Checksum::Store.new
-
       definition.resolve.sort_by(&:full_name).each do |spec|
-        out << (spec.source.checksum_store || empty_store)[spec].to_lock
+        checksums = spec.source.checksum_store&.checksums(spec.full_name)
```

Comment:
is checksum_store always non-nil now?

Reply:
I removed the `&` and running the test suite locally now to find out.